File: 0001-two-sum/main.go
func twoSum(nums []int, target int) []int {
	mp := make(map[int]int)
	for i, v := range nums {
		j, ok := mp[target-v]
		if ok {
			return []int{i, j}
		}
		mp[v] = i
	}
	return []int{-1, -1}
}

File: 0001-two-sum/main.cpp
class Solution {
public:
  vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int,int> lut;
    for (int i=0; i<nums.size(); ++i) {
      auto found = lut.find(target - nums[i]);
      if (found != lut.end()) {
        return {i, found->second};
      }
      lut[nums[i]] = i;
    }
    return {-1,-1};
  }
};